peer/brontide.go

```diff
@@ -886,6 +890,21 @@ func (p *Brontide) Start() error {
 		return fmt.Errorf("unable to load channels: %w", err)
 	}
 
+	onionMessageEndpoint := msgmux.NewOnionEndpoint(
+		p.cfg.OnionMessageServer,
```

Comment:
I am not sure if this is the right approach, we use 1 server-level Server with an update channel which is not buffered to funnel in all onion message from all peers, seem not really efficient ?

this is the update channel of the server:

https://github.com/lightningnetwork/lnd/blob/c6d6d4c0b6f3a3c921864253fc8e871878b93fc8/subscribe/subscribe.go#L154-L161

Reply:
Not sure if I completely follow you critique.

We have a single subscription endpoint for onion messages (similar to custom messages) which allows for "listening in" on onion messages received by this node. We pass along the subscription server here to the `msgmux` endpoint so that when we handle the onion message, we can notify a subscriber of an incoming message.

Buffering or rate limiting the actual onion message itself should happen upstream of the notification server, right?

Reply:
I think we talked about this via other means and agreed upon the fact that we need to differentiate incoming messages so that onion-messages will not hinder channel msges to be received in time.

Reply:
IIUC, this is just a stand in, as we don't have any forwarding behavior here yet. This is just the RPC server stream.